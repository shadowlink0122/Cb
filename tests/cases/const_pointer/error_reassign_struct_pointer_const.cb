// エラーテスト: Point* const の再代入
include "stdio.cb"

struct Point {
    int x;
    int y;
};

int main() {
    println("=== Test: Cannot reassign const pointer to struct ===");
    Point p1;
    Point p2;
    p1.x = 10;
    p1.y = 20;
    p2.x = 30;
    p2.y = 40;
    
    Point* const ptr = &p1;
    
    // エラー: Point* const 自体は変更できない
    ptr = &p2;  // This should fail
    
    println("ERROR: This should not execute");
    return 0;
}
