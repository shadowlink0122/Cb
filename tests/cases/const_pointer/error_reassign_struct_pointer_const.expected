=== Test: Cannot reassign const pointer to struct ===
Error: Cannot reassign const pointer (T* const)
